itou/www/apply/views/submit_views.py

```diff
@@ -245,6 +276,19 @@ def get(self, request, *args, **kwargs):
             else:
                 self.apply_session.init({"selected_jobs": [job_description.pk]})
 
+        # Go directly to step ApplicationJobsView if we're carrying the job seeker public id with us.
+        job_seeker, _, _ = _check_apply_for_job_seeker(self.request)
```

Comment:
Plutôt que de réutiliser `_check_apply_for_job_seeker` qui fait clairement plus que nécessaire, je dirais:
- soit de sortir un `get_job_seeker_to_apply_for(request)` qui ne fait que ce que tu souhaites et le mutualiser avec `_check_apply_for_job_seeker`
- soit directement mettre le code qui t'intéresse ici, comme pour le `if job_description_id := request.GET.get("job_description_id"):` juste au dessus.
  Un `if tunnel == "sender" and (job_seeker_public_id := request.GET.get("job_seeker")):` devrait faire l'affaire.